pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -865,98 +1060,111 @@ func (r *ReconcileRazeeDeployment) addFinalizer(razee *marketplacev1alpha1.Razee
 	return nil
 }
 
-func (r *ReconcileRazeeDeployment) MakeRazeeClusterMetaData(namespace, uuid string) *corev1.ConfigMap {
-
+func (r *ReconcileRazeeDeployment) MakeRazeeClusterMetaData(instance *marketplacev1alpha1.RazeeDeployment) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "razee-cluster-metadata",
-			Namespace: namespace,
+			Namespace: *instance.Spec.TargetNamespace,
 			Labels: map[string]string{
 				"razee/cluster-metadata": "true",
 				"razee/watch-resource":   "lite",
 			},
 		},
-		Data: map[string]string{"name": uuid},
+		Data: map[string]string{"name": instance.Spec.ClusterUUID},
 	}
 }
 
 //watch-keeper-non-namespace
 func (r *ReconcileRazeeDeployment) MakeWatchKeeperNonNamespace(
-	namespace string,
+	instance *marketplacev1alpha1.RazeeDeployment,
 ) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "watch-keeper-non-namespaced",
-			Namespace: namespace,
+			Namespace: *instance.Spec.TargetNamespace,
 		},
 		Data: map[string]string{"v1_namespace": "true"},
 	}
 }
 
 //watch-keeper-non-namespace
 func (r *ReconcileRazeeDeployment) MakeWatchKeeperLimitPoll(
-	namespace string,
+	instance *marketplacev1alpha1.RazeeDeployment,
 ) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "watch-keeper-limit-poll",
-			Namespace: namespace,
+			Namespace: *instance.Spec.TargetNamespace,
 		},
 	}
 }
 
-func (r *ReconcileRazeeDeployment) MakeWatchKeeperConfig(
-	namespace string,
-	rhmOperatorSecretValues RhmOperatorSecretValues,
-) *corev1.ConfigMap {
+//DeploySecretValues[RAZEE_DASH_URL_FIELD]
+func (r *ReconcileRazeeDeployment) MakeWatchKeeperConfig(instance *marketplacev1alpha1.RazeeDeployment) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "watch-keeper-config",
-			Namespace: namespace,
+			Namespace: *instance.Spec.TargetNamespace,
 		},
-		Data: map[string]string{"RAZEEDASH_URL": rhmOperatorSecretValues.razeeDashUrl, "START_DELAY_MAX": "0"},
+		Data: map[string]string{"RAZEEDASH_URL": instance.Spec.DeployConfig.RazeeDashUrl, "START_DELAY_MAX": "0"},
+	}
+}
+
+// Uses the SecretKeySelector struct to to retrieve byte data from a specified key
+func (r *ReconcileRazeeDeployment) GetDataFromRhmSecret(request reconcile.Request, sel corev1.SecretKeySelector) ([]byte, error) {
```

Comment:
Duplicated with utils func? Maybe keep this one and remove it from utils? Hard coded is fine here.

Reply:
yep I was thinking about exporting from utils at one point.